### DISCUSSION TITLE
how to execute serverless function which needs to call an authorized api?

### DISCUSSION THREAD

**Discussion by stardvst** [vercel/next.js#32144] [Help] [answered]
i have a serverless function (`/api/poll`), which should be executed by github actions.
i'm using next-auth as oauth provider, and store access tokens given by them in database.
in the function i call an external api, and i need to provide an access token. where should i get the access token from?
for user-specific things, i get access token from the db.
but i think the serverless function shouldn't be tied to a specific user, so i was thinking of using access token of the admin.
is this correct approach?

**Comment by tpetersons** [comment, marked as answer]
You said you are calling external API which requires an access token.
This means you have to look at their API documentation to see what are the requirements.
Most cases you will have some API key that you can put into your .env file and use it to get access token.
Sometimes you have to use client id and client secret. It really depends on external API that you are using.
Can you share which one is it?

**Reply by stardvst** [reply]
i'm using spotify oauth provider.
i do store the client id and client secret in .env file, and configure the provider with those values.

**Reply by toms-petersons-visma** [reply]
In this case you need to use client credentials flow: https://developer.spotify.com/documentation/general/guides/authorization/client-credentials/.
I suggest to find some spotify react library that handles authorization for this. This is a bit different from next auth library since you want to use API not related to user authentication.
So you will still use next auth for you current features, but for server side features you cannot use next auth.

**Reply by stardvst** [reply]
okay i get the idea, will have a look at credentials flow. thanks
